src/l500/l500-options.cpp

```diff
@@ -744,6 +777,8 @@ namespace librealsense
 
     void sensor_mode_option::set(float value)
     {
+        if (_value == value) return;
```

Comment:
1. Is it possible that the value is volatile in FW?
2.Consider switching to 
`std::abs( value - _value ) < std::numeric_limits< float >::epsilon()`

Reply:
1. What do you mean?
Sensor_mode is a SW option, FW is not in the loop, even if it wasn't I am not sure that this comment was not true to all options that can be changed from several threads.
2. Good idea

Reply:
1 - Got it